broadlink/climate.py

```diff
@@ -201,3 +201,282 @@ def set_schedule(self, weekday: List[dict], weekend: List[dict]) -> None:
             input_payload.append(int(weekend[i]['temp'] * 2))
 
         self.send_request(input_payload)
+
+class tornado(device):
+    """Controls a Tornado 16X SQ air conditioner."""
+    def __init__(self, *args, **kwargs):
+        device.__init__(self, *args, **kwargs)
+        self.type = "Tornado air conditioner"
+    
+    def _decode(self, response):
+        payload = self.decrypt(bytes(response[0x38:]))
+        return payload
+    
+    def _calculate_checksum(self, arr, target=0x20017, checksum=None):
+        """Calculate checksum of given array,
+        by adding little endian words and subtracting from target.
+        
+        Args:
+            arr (list): 
+            checksum (Optional[int]):  a received checksum, can be used to calc target
+        """
+        diff = 0
+        for i in range(len(arr)): # might need an override
+            if (i % 2 == 0):
+                try:
+                    diff += (arr[i] + (arr[i+1] << 8))
+                except:
+                    # odd array size, add little value
+                    diff += arr[i]
+        
+        if (checksum):
+            print('checksum diff is {}'.format(diff + checksum))
+
+        result = target - (diff & 0xffff)
+        return [result & 0xff, (result >> 8) & 0xff]
```

Comment:
This function is not necessary. We use sum() for checksums.

Reply:
I rewrote it to use sum(), it's more complex than the other uses of sum() and I think having it as function is more readable. What do you think?

Reply:
How about [this](https://github.com/mjg59/python-broadlink/pull/430#discussion_r494007259)?

Reply:
How about [this](https://github.com/mjg59/python-broadlink/pull/430#discussion_r494007259)?